thesis-gui/src/components/my-account/UserRanking.tsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import {
+    IonCard,
+    IonCardContent,
+    IonCardSubtitle,
+    IonCardTitle,
+    IonItem,
+    IonLabel,
+    IonList,
+    IonRouterLink
+} from "@ionic/react";
+import {useUserContext} from "../../context/UserContext";
+import {useCategoryContext} from "../../context/CategoryContext";
+
+const CategoryRanksItem: React.FC<{ name: string, rank: number }> = ({ name, rank }) => {
+    return (
+        <IonItem>
+            <IonLabel>{ name }</IonLabel>
+            <IonLabel slot="end">
+                { rank }
+            </IonLabel>
+        </IonItem>
+    );
+}
+
+export const UserRanking = () => {
+    const { user } = useUserContext();
+    const { categories } = useCategoryContext();
+
+    const getCategoryNameWithId = (id: number) => {
```

Comment:
I think this whole function can be simplified to:
`return categories.find(category => category.id === id)?.name || '';`

Reply:
Done